### ISSUE TITLE
If element is initially hidden with v-show signaturePad doesn't work

### ISSUE THREAD

**Issue by garhbod** [neighborhood999/vue-signature-pad#81]
Because v-show hides/removes the element completely from the DOM then `new SignaturePad(...` can never find the element to add watchers.

Maybe some of the `mount()` functionality can be moved to a seperate method and be recalled as required

**Comment by garhbod** [comment]
Solved the issue by running the following when the element became visible again.
`this.$refs.signaturePad.resizeCanvas(this.$refs.signaturePad.$refs.signaturePadCanvas);`

I will create a pull request so that this line can be simplified to
`this.$refs.signaturePad.resizeCanvas();`

**Comment by garhbod** [comment]
Can't get yarn to install dependencies so I'll just put what I was going to change here

```
  methods: {
    resizeCanvas() {
      const canvas = this.$refs.signaturePadCanvas;
      const data = this.signaturePad.toData();
      const ratio = Math.max(window.devicePixelRatio || 1, 1);
      canvas.width = canvas.offsetWidth * ratio;
      canvas.height = canvas.offsetHeight * ratio;
      canvas.getContext('2d').scale(ratio, ratio);
      this.signaturePad.clear();
      this.signatureData = TRANSPARENT_PNG;
      this.signaturePad.fromData(data);
    },
...
```
Or if the input is required some of the time
```
  methods: {
    resizeCanvas(inputCanvas) {
      const canvas = inputCanvas || this.$refs.signaturePadCanvas;
      const data = this.signaturePad.toData();
      const ratio = Math.max(window.devicePixelRatio || 1, 1);
      canvas.width = canvas.offsetWidth * ratio;
      canvas.height = canvas.offsetHeight * ratio;
      canvas.getContext('2d').scale(ratio, ratio);
      this.signaturePad.clear();
      this.signatureData = TRANSPARENT_PNG;
      this.signaturePad.fromData(data);
    },
...
```

**Comment by neighborhood999** [comment]
@garhbod Thanks for submitting this issue, I will check this problem when I have free time.

> Can't get yarn to install dependencies so I'll just put what I was going to change here

It's doesn't matter, you can submit PR directly. By the way, maybe you can create an example for reproducing this problem on [Codesanbox](http://codesandbox.io) :wink:

**Comment by neighborhood999** [comment]
@garhbod I read [conditional rendering](https://vuejs.org/v2/guide/conditional.html#v-show) of Vue docs, `v-show` isn't **hides/removes** elements, actually, `v-show`'s behavior:

> The difference is that an element with `v-show` will always be rendered and remain in the DOM; `v-show` only toggles the `display` CSS property of the element.

I create a demo on [Codesandbox](https://codesandbox.io): [v-show with resize](https://codesandbox.io/s/ym4voqw5q1).

**Comment by garhbod** [comment]
If it toggles the `display` CSS property than that is *hiding* it. 

Your example works because of the listeners that vue-signature-pad has already set on the resize event are being called also - `resizeCanvas`

I have forked your example with a show button so you can see this issue https://codesandbox.io/s/o49241xrlz

**Comment by garhbod** [comment]
After you click the show button you can see that the signature is not displaying as you draw.

Once you resize the window it appears.

Refresh and try the **Show with Resize** button and you will see it work

**Comment by neighborhood999** [comment]
@garhbod Nice! That's a google example for reproducing the problem, seems this may solve #54 and #62 problem.

**Comment by neighborhood999** [comment]
Done, Merged at #82
